src/extension/content-script/batteries/GitHub.ts

```diff
@@ -1,77 +1,125 @@
   const username = urlParts[1];
   const repo = urlParts[2];
 
+  if (username && !validationRegex.test(username)) return; // invalid username
+  if (username && repo && !validationRegex.test(repo)) return; // invalid repo
+
+  let lightningData = null;
+
+  // Search lightning data in the repository if the current page is a repository page
   if (username && repo) {
-    handleRepositoryPage(username);
-  } else if (username) {
-    handleProfilePage();
+    lightningData = await getLightningDataFromRepository(username, repo);
   }
-};
 
-function parseElement(elementSelector: string) {
-  const text = document.querySelector<HTMLElement>(elementSelector)?.innerText;
-  if (text) {
-    const match = text.match(/(⚡:?|lightning:|lnurl:)\s?(\S+@\S+)/i);
-    if (match) return match[2];
+  // Search lightning data in the user profile or organization profile if the current page is a profile page
+  // or if the current page is a repository page but the lightning data is not set at repository level
+  if (!lightningData && username) {
+    lightningData = await getLightningDataFromProfile(username);
   }
+
+  if (lightningData) setLightningData([lightningData]);
+};
+
+function findLightningAddressInText(text: string) {
+  const match = text.match(/(⚡:?|lightning:|lnurl:)\s?(\S+@\S+)/i);
+  if (match) return match[2];
 }
 
-function handleProfilePage() {
-  const shortBioElement = document.querySelector<HTMLHeadingElement>(
-    ".Layout-sidebar .h-card [data-bio-text]"
-  );
-
-  // This is not a GitHub profile
-  if (!shortBioElement) return;
-
-  const address =
-    parseElement(".Layout-sidebar .h-card [data-bio-text]") ||
-    parseElement(".Layout-main article");
-
-  if (!address) return;
-
-  setLightningData([
-    {
-      method: "lnurl",
-      recipient: address,
-      ...getOriginData(),
-      description: shortBioElement?.innerText ?? "",
-      name:
-        document.querySelector<HTMLHeadingElement>(
-          '.Layout-sidebar .h-card [itemprop="name"]'
-        )?.innerText ?? "",
-      icon:
-        document.querySelector<HTMLImageElement>(
-          `.Layout-sidebar .h-card img.avatar-user`
-        )?.src ?? "",
-    },
-  ]);
+async function getLightningDataFromProfile(
+  username: string
+): Promise<Battery | null> {
+  const userProfile = await fetch(
+    // fetch user data
+    `https://api.github.com/users/${username}`
+  ).then((res) => res.json());
+
+  if (!userProfile || !userProfile.login) return null; // not an user profile
+
+  let address = null;
+
+  if (userProfile.bio) {
+    // search in bio
+    address = findLightningAddressInText(userProfile.bio);
+  }
+
+  if (!address) {
+    // search in profile's readme
+    const userRepoReadmeData = await fetch(
+      `https://api.github.com/repos/${username}/${username}/readme`
+    ).then((res) => res.json());
+
+    if (userRepoReadmeData.download_url) {
+      const userRepoReadmeContent = await fetch(
```

Comment:
I think instead of downloading the file we already get the content of that file in `userRepoReadmeData.content`. 
it is a base64 encoded string.

but we need to think that anything can be in there, the user controls the repo an could upload any file.
maybe we just check the bio for now?